README.md

```diff
@@ -2,7 +2,251 @@
+    Click me
+</button>
+```
+Next to the modifiers that you can see are being set here, like `padding--xs`, 
+the component also has a number of default modifiers.
+As you might have expected from StandardKit, you can override / configure this. For example:
+```scss
+$components: (
+  button: (
+    default-color: primary,
+    default-margin: xl,
+    default-padding: xs
+  )
+);
+```
+Now, you can just use:
+```html
+<div class="button">
+    I am a configured button
+</div>
+```
+This becomes useful as you have more buttons scattered through-out your application.
+
+
+### Types
+Most of the time, applications tend to have multiple definitions of a button. 
+For example, we might have a `cta-button`, a `save-button` and a `cancel-button`.
+Using just the component configuration, you can only optimize for one situation. With types, we don't have to.
+##### Types are a compound component definition that combines multiple modifiers on top of the default modifiers
+For example:
+```scss
+$components: (
+  button: (
+    default-color: primary,
+    cta: ( default-color: custom, default-margin: xl),
+    save: ( default-padding: xs),
+    cancel: ( default-color: error )
+  )
+);
+```
+To now use these three buttons, I can just:
+```html
+<button class="button--cta">Call-to-Action</button>
+
+<button class="button--save">Save</button>
+
+<button class="button--cancel">Cancel</button>
+```
+Very simple yet powerful.
+
+
+### Special Cases
+A couple of modifiers are special.
+#### Color & Fill Combo
+This is relevant because `color` depends on `fill`. <br> 
+For instance, with a `fill: 'fill'` the `text` color is the `contrast` color and the `background` is the `main` color.  
+<sub>See `Color Generation` for more information about colors and their (generated) subsets.</sub>
+
+#### Complex Modifiers
+Some modifiers can be configured a little bit more than others. For example:
+```scss
+// Colors
+$colors: (
+  primary: #fff,
+  secondary: (main: #fff, complement: #f00, contrast: #000, complement-contrast: #0ff )
+);
+
+// Elevation
+$elevation: (
+  high: 20,
+  custom: () // TODO : Implement 
+);
+```
+<sub>In this case all values are optional, but you need at least `main` or `complement` to generate the others.</sub>
+
+#### Breakpoints
+Breakpoints is a configurable modifiers, you can add or change them to your liking. <br> 
+By default it's using `em` but feel free to change to what you want.
+
+However, sometimes you want to change the behaviour of other modifiers based on the viewport. For example:
+```scss
+$margins: (
+  custom: ( start: square, s: 3rem, l: round )
+)
+```
+As you can see, something weird is going on. 
+I'm using `round` which is one of the flavors, but I'm also specifying `3rem`, an actual value. 
+This always looks for the flavor first. So it would be unwise to call a flavor by one of the values.
+
+StandardKit uses the `start` keyword to distinguish if this is a breakpoint-scenario. 
```

Comment:
Maybe mention something about the breakpoints following the "mobile-first" principle.